_posts/2017-11-21-dotnet-core-rest-api-in-about-30-minutes.md

```diff
@@ -0,0 +1,587 @@
+
+  ```c#
+  // PUT api/airplanes/12345/fly
+  [HttpPut("{sn}/fly")]
+  public async Task Fly(string sn, [FromBody]TimeSpan estimatedTripTime)
+  {
+      var airplane = _context.Airplanes.Where(x => x.SerialNumber == sn).FirstOrDefault();
+      if (airplane != null)
+      {
+          airplane.Fly(estimatedTripTime);
+      }
+
+      await _context.SaveChangesAsync();
+  }
+
+  // PUT api/airplanes/12345/land
+  [HttpPut("{sn}/land")]
+  public async Task Land(string sn, [FromBody]string airportCode)
+  {
+      var airplane = _context.Airplanes.Where(x => x.SerialNumber == sn).FirstOrDefault();
+      if (airplane != null)
+      {
+          airplane.Land(airportCode);
+      }
+
+      await _context.SaveChangesAsync();
+  }
+  ```
+
+  Let's try it out.
+  1) Restart the app and perform the `POST` again
+  2) Use Postman to make a `PUT` call to `http://localhost:5000/api/airplanes/12345/fly` with the request body simply containing the `json`: `"00:30:00"`.
+  3) Now, if you do a `GET`, you should see that the `lastTakeoffTime` is set to the time you made the request and the `estimatedLandingTime` is set to 30 minutes later.
+  4) Use Postman to make a `PUT` call to `http://localhost:5000/api/airplanes/12345/land` with the request body simply containing the `json`: `"JFK"`.
+  5) Now, the times should both be `null` and the `currentAirportCode` is set to JFK! This demonstrates the advantages of using an Object Oriented approach. We've allowed the client to only care about one piece of data for each `PUT` request, but have modified up to 3 different properties correctly, rather than relying on the client to do that work!
+
+***
+* DELETE
+
+  Finally, the `DELETE` method is pretty straightforward:
+
+  ```c#
+  // DELETE api/airplanes/12345
+  [HttpDelete("{sn}")]
+  public async Task Delete(string sn)
+  {
+      var airplane = _context.Airplanes.Where(x => x.SerialNumber == sn).FirstOrDefault();
+      if (airplane != null)
+      {
+          _context.Airplanes.Remove(airplane);
+      }
+
+      await _context.SaveChangesAsync();
+  }
+  ```
+
+  Try it out by restarting the app, performing the `POST` (so you have something to delete), then have Postman hit the `DELETE` method of the URL: `http://localhost:5000/api/airplanes/12345`. Now if you perform a `GET` for that record, you should receive a `204 - No Content`.
+
+  ***
+  :thought_balloon:
+
+  Based on [good REST patterns](https://restpatterns.mindtouch.us/HTTP_Status_Codes), `204 - No Content` is really not want we want to see for the `GET` of a missing record. We really want to see `404 - Not Found`, but we'll have to modify it to return an `IActionResult`, which we will go over in a later post.
+  ***
```

Comment:
in the output i see `posts. ***`

Reply:
Wow, I wonder why it's formatting the markdown so much differently than both VS Code Preview and TFS Preview... I'll have to just go through again and fix all these little formatting issues.